sphinx/api/configuration/file-creation.rst

```diff
@@ -25,6 +25,19 @@ root directory to install file targets with a relative path:
       :opt svn: The Subversion scheme. The location is a Subversion URL or
          an FCM location keyword. A URI with these schemes ``svn``,
          ``svn+ssh`` and ``fcm`` are automatically recognised.
+      :opt git: The Git scheme. The location is complex due to Git semantics.
+         It must have the scheme ``git`` and be of the form
+         ``git:REPOSITORY_URL::PATHSPEC::TREEISH``. ``REPOSITORY_URL`` should
+         be a Git repository URI which may itself have a scheme ``ssh``,
+         ``git``, ``https``, or be of the form ``HOST:PATH``, or ``PATH`` for
+         local repositories. ``PATHSPEC`` should be a path to a file or
+         directory that you want to extract. ``TREEISH`` should be a tag,
+         branch, or long commit hash to specify the commit at which you want
+         to extract. These should follow the same semantics as if you git
+         cloned ``REPOSITORY_URL``, git checkout'ed ``TREEISH``, and extracted
+         the path ``PATHSPEC`` within the clone. To extract from the root of
+         the repository use a ``PATHSPEC`` of ``./`` e.g.
+         ``git:git@github.com:metomi/rose::./::2.2.0``.
```

Comment:
The file install docs could really do with a couple of examples, especially for Git as the syntax is quite different to that used by the Git CLI:

```
git -C <protocol>:<path> show <ref>:<file>

source=git:<protocol>:<path>::<file>::<ref>
```

And it's tempting for users to mistake the `git` schema for the `git` protocol or simply replace it with the protocol.

Reply:
Agreed, expanded a bit.